.github/workflows/ci.yaml

```diff
@@ -0,0 +1,33 @@
+name: CI
+
+on:
+  - pull_request
+  - push
+
+concurrency:
+  group: ${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: Ruby
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby:
+          - "2.7"
+          - "3.0"
+          - "3.1"
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
```

Comment:
I've implemented both the matrix and code checkout + ruby setup inside the reusable workflow. https://github.com/theforeman/actions#gem-test should be the complete example, except that you should specify the Ruby versions via `with` like you also specified `test`.

Reply:
This is now done and https://github.com/theforeman/actions#gem-test is now the complete example (except for the test command). It automatically detects the compatible Ruby versions.